packages/perseus/src/widgets/expression/expression.test.tsx

```diff
@@ -523,4 +523,33 @@ describe("Expression Widget", function () {
             ).toBeNull();
         });
     });
+
+    describe("propUpgrades", () => {
+        it("can upgrade from v0 to v1", () => {
+            const v0props = {
+                times: false,
+                buttonSets: ["basic"],
+                functions: [],
+                form: false,
+                simplify: false,
+                value: "42",
+            };
+
+            const result = ExpressionWidgetExport.propUpgrades["1"](v0props);
```

Comment:
**suggestion**: Typically, I'm against explicit typing for variables, but in this case I think it's a good additional validation to type this `result` variable as `PerseusExpressionWidgetOptions` so that TypeScript ensures that the prop upgrade it typed correctly.

Reply:
I went back and forth on this. This won't work:

``` TS
const result: WhateverWidgetOptions = WhateverWidgetExport.propUpgrades["1"](v0props);
```

...because the `propUpgrades` function takes `any` and modifies it (which basically makes the return value `any`). Also because 2/3 `propUpgrades` returned `any` (I just changed that).

So I did a mix:

``` TS
const expected: WhateverWidgetOptions = {}
const result: WhateverWidgetOptions = WhateverWidgetExport.propUpgrades["1"](v0props);
expect(result).toEqual(expected)
```

Not sure how helpful it'll be though.